test/skimNtuple_HHbtag.cpp

```diff
@@ -5426,67 +5368,110 @@ int main (int argc, char** argv)
+						  theSmallTree.m_subjetjet1_bID_deepFlavor  = theBigTree.subjets_deepFlavor_probb->at(isj) + theBigTree.subjets_deepFlavor_probbb->at(isj) + theBigTree.subjets_deepFlavor_problepb->at(isj) ;
+						}
+					  else if (nSJ == 2)
+						{
+						  tlv_subj2.SetPxPyPzE (theBigTree.subjets_px->at(isj), theBigTree.subjets_py->at(isj), theBigTree.subjets_pz->at(isj), theBigTree.subjets_e->at(isj));
+						  theSmallTree.m_subjetjet2_pt   = tlv_subj2.Pt();
+						  theSmallTree.m_subjetjet2_eta  = tlv_subj2.Eta();
+						  theSmallTree.m_subjetjet2_phi  = tlv_subj2.Phi();
+						  theSmallTree.m_subjetjet2_e    = tlv_subj2.E();
+						  theSmallTree.m_subjetjet2_bID  = theBigTree.subjets_CSV->at(isj) ;
+						  theSmallTree.m_subjetjet2_bID_deepCSV  = theBigTree.subjets_deepCSV_probb->at(isj) + theBigTree.subjets_deepCSV_probbb->at(isj) ;
+						  theSmallTree.m_subjetjet2_bID_deepFlavor  = theBigTree.subjets_deepFlavor_probb->at(isj) + theBigTree.subjets_deepFlavor_probbb->at(isj) + theBigTree.subjets_deepFlavor_problepb->at(isj);
+						}
+					}
+
+				  if (tlv_firstBjet.Mag2()==0 or tlv_secondBjet.Mag2()==0) {
+					throw std::runtime_error("[ERROR] The b-jets were not defined.");
+				  }
+				  
+				  bool A1B2 = (tlv_subj1.DeltaR(tlv_firstBjet)  < 0.4) and (tlv_subj2.DeltaR(tlv_secondBjet) < 0.4);
+				  bool A2B1 = (tlv_subj1.DeltaR(tlv_secondBjet) < 0.4) and (tlv_subj2.DeltaR(tlv_firstBjet)  < 0.4);
 			            
-			      if(DEBUG)
-				{
-				  cout << " fatjet: idx " << fjIdx << " nsj=" << sjIdxs.size()
-				       << " sj1pt=" << tlv_subj1.Pt() << " sj1eta=" << tlv_subj1.Eta() << " sj1phi=" << tlv_subj1.Phi()
-				       << " sj2pt=" << tlv_subj2.Pt() << " sj2eta=" << tlv_subj2.Eta() << " sj2phi=" << tlv_subj2.Phi()
-				       << " passMatch=" << (A1B2 || A2B1) << endl;
-				}
+				  if(DEBUG)
+					{
+					  cout << " fatjet: idx " << fjIdx << " nsj=" << sjIdxs.size()
+						   << " sj1pt=" << tlv_subj1.Pt() << " sj1eta=" << tlv_subj1.Eta() << " sj1phi=" << tlv_subj1.Phi()
+						   << " sj2pt=" << tlv_subj2.Pt() << " sj2eta=" << tlv_subj2.Eta() << " sj2phi=" << tlv_subj2.Phi()
+						   << " passMatch=" << (A1B2 || A2B1) << endl;
+					}
 			            
-			      theSmallTree.m_dR_subj1_subj2 = tlv_subj1.DeltaR(tlv_subj2);
+				  theSmallTree.m_dR_subj1_subj2 = tlv_subj1.DeltaR(tlv_subj2);
 			            
-			      if (A1B2 || A2B1) theSmallTree.m_fatjet_hasMatchedSj = true;  //has two sub-jets matched to the resolved jets
+				  if (A1B2 || A2B1) theSmallTree.m_fatjet_hasMatchedSj = true;  //has two sub-jets matched to the resolved jets
 			            
-			    } 
-			} // end if fatjet_bTag not empty 
-		    } // end if (theBigTree.ak8jets_px->size() > 0)  (end of the boosted section)
-		} // end if (jets_and_sortPar.size () >= 2) 
+				}
 
+			  // systematics for AK8 jets
+			  pair<vector<double>, vector<double>> unc_AK8_updown = JECprovider.getJECUncVectors(fjIdx, theBigTree);
```

Comment:
The source code of JECProvider must be modified to handle different AK4 and AK8 jets. 
The corresponding weight files needs to be added to the repository.

Reply:
Me and @kramerto discussed this. We are currently using the "RegroupedV2" files, which do not contain AK8 corrections, only AK4 (see [this Twiki](https://twiki.cern.ch/twiki/bin/view/CMS/JECDataMC#Jet_Energy_Corrections_in_Run2)). I suppose the term "regrouping" refers to the [reduced set of JECs](https://twiki.cern.ch/twiki/bin/view/CMS/JECUncertaintySources#Run_2_reduced_set_of_uncertainty) we are considering. I'm unsure on how to proceed for AK8 jets.

Reply:
In the same twiki you linked the are the txt files with corrections for AK8 so I think we should define new vector of uncertainties specific for AK8 using these files.

Reply:
The conveners [confirmed](https://cms-talk.web.cern.ch/t/jecs-for-ak8/36525/3) we should we the same uncertainties as for AK4 jets.